### PR TITLE
Improve TLS docs

### LINKED ISSUE #1566: Add guidance about how to skip mounting the TLS secret for the TiDB client
## Change Request

**This repository is ONLY used to solve issues related to TiDB Operator documentation.
For other issues (related to TiDB, PD, etc), please move to [other repositories](https://github.com/pingcap/).**

Please answer the following questions before submitting your issue. Thanks!
https://asktug.com/t/topic/303095/4
1. Describe what you find is inappropriate or missing in the existing docs.

2. Describe your suggestion or addition.

3. Provide some reference materials (documents, websites, etc) if you could.

### LINKED ISSUE #1607: TLS doc improvement
## Change Request

**This repository is ONLY used to solve issues related to TiDB Operator documentation.
For other issues (related to TiDB, PD, etc), please move to [other repositories](https://github.com/pingcap/).**

Please answer the following questions before submitting your issue. Thanks!

1. Describe what you find is inappropriate or missing in the existing docs.
We have to make it clear for the behavior of setting `spec.tlsCluster.enabled: true` after the cluster has been created.
Refer to https://asktug.com/t/topic/512910.
2. Describe your suggestion or addition.

3. Provide some reference materials (documents, websites, etc) if you could.

### PR DESCRIPTION
**Pull request by just1900**
Signed-off-by: just1900 <legendj228@gmail.com>

### What is changed, added, or deleted? (Required)

fix #1607 fix #1566 

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.3 (TiDB Operator 1.3 versions)
- [ ] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):

### CODE REVIEW

File: zh/enable-tls-for-mysql-client.md [hunk 1]
```diff
@@ -14,7 +14,16 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-for-mysql-client/']
     >
     > 创建的 Secret 对象必须符合上述命名规范，否则将导致 TiDB 集群部署失败。
 
-2. 部署集群，设置 `.spec.tidb.tlsClient.enabled` 属性为 `true`；
+2. 部署集群，设置 `.spec.tidb.tlsClient.enabled` 属性为 `true`：
+
+    * 如果希望关闭 TiDB 服务端对客户端 CA 证书的认证，即在[配置 TiDB 服务端启用安全连接](https://docs.pingcap.com/tidb/stable/enable-tls-between-clients-and-servers#configure-tidb-server-to-use-secure-connections) 中不设置 ssl-ca 参数，可以设置 `.spec.tidb.tlsClient.disableClientAuthn` 属性为 `true`。
+    * 如果希望跳过作为 MySQL 客户端的内部组件(如 TidbInitializer, Dashboard, Backup, Restore)的 CA 证书认证，可以设置 `.spec.tidb.tlsClient.skipInternalClientCA` 属性为 `true`。
+    * 如果希望跳过作为 MySQL 客户端的内部组件的 TLS 认证，可以给集群对应的 `TidbCluster` 加上 `tidb.tidb.pingcap.com/skip-tls-when-connect-tidb=“true”` 的 annotation。
```
Comment (csuzhangxc) [inline zh/enable-tls-for-mysql-client.md:21]:
```suggestion
    * 如果希望跳过作为 MySQL 客户端的内部组件的 TLS 认证，可以给集群对应的 `TidbCluster` 加上 `tidb.tidb.pingcap.com/skip-tls-when-connect-tidb="true"` 的 annotation。
```

**Review by csuzhangxc — APPROVED** [review]

File: zh/enable-tls-for-mysql-client.md [hunk 2]
```diff
@@ -14,7 +14,16 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-for-mysql-client/']
     >
     > 创建的 Secret 对象必须符合上述命名规范，否则将导致 TiDB 集群部署失败。
 
-2. 部署集群，设置 `.spec.tidb.tlsClient.enabled` 属性为 `true`；
+2. 部署集群，设置 `.spec.tidb.tlsClient.enabled` 属性为 `true`：
+
+    * 如果希望关闭 TiDB 服务端对客户端 CA 证书的认证，即在[配置 TiDB 服务端启用安全连接](https://docs.pingcap.com/tidb/stable/enable-tls-between-clients-and-servers#configure-tidb-server-to-use-secure-connections) 中不设置 ssl-ca 参数，可以设置 `.spec.tidb.tlsClient.disableClientAuthn` 属性为 `true`。
+    * 如果希望跳过作为 MySQL 客户端的内部组件(如 TidbInitializer, Dashboard, Backup, Restore)的 CA 证书认证，可以设置 `.spec.tidb.tlsClient.skipInternalClientCA` 属性为 `true`。
+    * 如果希望跳过作为 MySQL 客户端的内部组件的 TLS 认证，可以给集群对应的 `TidbCluster` 加上 `tidb.tidb.pingcap.com/skip-tls-when-connect-tidb="true"` 的 annotation。
```
Comment (DanielZhangQD) [inline zh/enable-tls-for-mysql-client.md:21]: 加一下从哪个版本开始支持

File: zh/enable-tls-for-mysql-client.md [hunk 3]
```diff
@@ -14,7 +14,16 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-for-mysql-client/']
...
+
+    > **注意：**
+    >
+    > 为已部署的集群设置 `.spec.tidb.tlsClient.enabled` 属性为 `true`，将导致 TiDB Pod 重启。
```
Comment (DanielZhangQD) [inline zh/enable-tls-for-mysql-client.md:25]:
```suggestion
    > 已部署的集群 `.spec.tidb.tlsClient.enabled` 属性从 `false` 改为 `true`，将导致 TiDB Pod 滚动重启。
```

**Review by csuzhangxc — APPROVED** [review]

File: zh/enable-tls-for-mysql-client.md [hunk 4]
```diff
@@ -14,7 +14,16 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-for-mysql-client/']
     >
     > 创建的 Secret 对象必须符合上述命名规范，否则将导致 TiDB 集群部署失败。
 
-2. 部署集群，设置 `.spec.tidb.tlsClient.enabled` 属性为 `true`；
+2. 部署集群，设置 `.spec.tidb.tlsClient.enabled` 属性为 `true`：
+
+    * 对于 TiDB Operator v1.1.0 及以后的版本，你可以给集群对应的 `TidbCluster` 加上 `tidb.tidb.pingcap.com/skip-tls-when-connect-tidb="true"` 的 annotation，来跳过作为 MySQL 客户端的内部组件(如 TidbInitializer, Dashboard, Backup, Restore)的 TLS 认证。
+    * 对于 TiDB Operator v1.3.0 及以后的版本，你可以设置 `.spec.tidb.tlsClient.disableClientAuthn` 属性为 `true`，来关闭 TiDB 服务端对客户端 CA 证书的认证，即在[配置 TiDB 服务端启用安全连接](https://docs.pingcap.com/tidb/stable/enable-tls-between-clients-and-servers#configure-tidb-server-to-use-secure-connections) 中不设置 ssl-ca 参数。
+    * 对于 TiDB Operator v1.3.0 及以后的版本，你可以设置 `.spec.tidb.tlsClient.skipInternalClientCA` 属性为 `true`，来跳过作为 MySQL 客户端的内部组件(如 TidbInitializer, Dashboard, Backup, Restore)的 CA 证书认证。
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md:21]:
```suggestion
    * 如需跳过作为 MySQL 客户端的内部组件（如 TidbInitializer、Dashboard、Backup、Restore）的 TLS 认证，你可以给集群对应的 `TidbCluster` 加上 `tidb.tidb.pingcap.com/skip-tls-when-connect-tidb="true"` 的 annotation。
    * 如需关闭 TiDB 服务端对客户端 CA 证书的认证，你可以设置 `.spec.tidb.tlsClient.disableClientAuthn` 属性为 `true`，即在[配置 TiDB 服务端启用安全连接](https://docs.pingcap.com/tidb/stable/enable-tls-between-clients-and-servers#configure-tidb-server-to-use-secure-connections) 中不设置 ssl-ca 参数。
    * 如需跳过作为 MySQL 客户端的内部组件（如 TidbInitializer、Dashboard、Backup、Restore）的 CA 证书认证，你可以设置 `.spec.tidb.tlsClient.skipInternalClientCA` 属性为 `true`。
```
这个 PR 只 cp 到 1.3 的话，可以不用说明版本了

**Review by ran-huang — APPROVED** [review]
Rest LGTM

**Pull request merged**